classes/security/Validation.php

```diff
@@ -474,6 +482,65 @@ public static function canAdminister($administeredUserId, $administratorUserId)
         // There were no conflicting roles. Permit administration.
         return true;
     }
+
+    /**
+     * Get the user's administration level
+     *
+     * @param int   $administeredUserId     User ID of user to potentially administer
+     * @param int   $administratorUserId    User ID of user who wants to do the administrating
+     * 
+     * @return int The authorized adminstration level 
+     */
+    public static function getAdministrationLevel($administeredUserId, $administratorUserId)
+    {
+        $roleDao = DAORegistry::getDAO('RoleDAO'); /** @var RoleDAO $roleDao */
+        $context = Application::get()?->getRequest()?->getContext();
+
+        // You can administer yourself
+        if ($administeredUserId == $administratorUserId) {
+            return self::ADMINISTRATION_FULL;
+        }
+
+        // You cannot adminster administrators
+        if ($roleDao->userHasRole(\PKP\core\PKPApplication::CONTEXT_SITE, $administeredUserId, Role::ROLE_ID_SITE_ADMIN)) {
+            return self::ADMINISTRATION_PROHIBITED;
+        }
+
+        // Otherwise, administrators can administer everyone
+        if ($roleDao->userHasRole(\PKP\core\PKPApplication::CONTEXT_SITE, $administratorUserId, Role::ROLE_ID_SITE_ADMIN)) {
+            return self::ADMINISTRATION_FULL;
+        }
+
+        // Check for administered user group assignments in other contexts
+        // that the administrator user doesn't have a manager role in.
+        $userGroups = Repo::userGroup()->userUserGroups($administeredUserId);
+        foreach ($userGroups as $userGroup) {
+            if ($userGroup->getContextId() != \PKP\core\PKPApplication::CONTEXT_SITE && !$roleDao->userHasRole($userGroup->getContextId(), $administratorUserId, Role::ROLE_ID_MANAGER)) {
+                // Found an assignment: disqualified.
+                // But also determine if a partial administrate is allowed
+                // if the Administrator User is a Journal Manager in the current context
+                if ($context && $roleDao->userHasRole($context->getId(), $administratorUserId, Role::ROLE_ID_MANAGER)) {
+                    return self::ADMINISTRATION_PARTIAL;
+                }
+                return self::ADMINISTRATION_PROHIBITED;
+            }
+        }
+
+        // Make sure the administering user has a manager role somewhere
+        $foundManagerRole = false;
+        $roles = $roleDao->getByUserId($administratorUserId);
+        foreach ($roles as $role) {
```

Comment:
This could be rewritten using the `UserGroup` collector using e.g. (off the top of my head):
```php
Repo::userGroup()->collector()
    ->filterByUserIds([$administratorUserId])
    ->filterByRoleIds([ROLE_ID_MANAGER])->count()
```

Reply:
done .